### PR TITLE
Support emulated aarch64 builds

### PR DESCRIPTION
**Pull request by dreamlayers**
A new class, Builder_emuarm64 handles this in prepare_cloud_image.
Now .kitchen.yml is created when preparing the image, and there is
only one VM listed, corresponding to the prepared image. Then
buildbot simply assumes there is only one VM and runs that, without
detecting architecture.

-------------

I hope I was reasonably respectful to the object oriented architecture of prepare_cloud_image.py, and I hope I didn't break the native ARM build, which I cannot test.

### CODE REVIEW

File: buildbot [hunk 1]
```diff
@@ -14,8 +14,8 @@ def echo_run(cmd):
 
 @contextmanager
 def instance():
-    arch = run(['uname', '-m'], stdout=PIPE).stdout.decode('latin1').strip()
-    vm = 'vm-{}'.format(arch).replace('_', '-')
+    # Unnecessary when there is only one VM, but allows yield and less changes
```
Comment (mgax) [inline buildbot]: I understand what you're doing here, and I think we can move this comment from the source code, into an extended git commit message. If anyone wants to dig into the history of that line, they can `git blame`, and should find that git commit message :)

File: prepare_cloud_image.py [hunk 2]
```diff
@@ -83,8 +84,35 @@ def build(self):
...
         self.prepare_disk_image()
         self.create_cloud_init_image()
+        self.create_kitchen_yml()
         self.run_qemu()
 
+    def create_kitchen_yml(self, name, extra = ()):
+        with open('.kitchen.yml', 'w') as f:
+            print(f)
+            f.write(
+                'driver:\n' +
```
Comment (mgax) [inline prepare_cloud_image.py]: Can we extract this as a template string using triple-quote string syntax? e.g. [this hostapd.conf template](https://github.com/liquidinvestigations/setup/blob/master/ansible/roles/hotspot/templates/boot.py#L11-L30)

**Review by mgax — CHANGES_REQUESTED** [review]
The changes look sane, I haven't tested them but I'm sure you did :)

**Comment by dreamlayers** [comment]
Okay, I force-pushed an amended commit to the pull request branch. I hope that's okay inside pull requests. It makes history in the main repo cleaner.

**Comment by mgax** [comment]
> Okay, I force-pushed an amended commit to the pull request branch. I hope that's okay inside pull requests. It makes history in the main repo cleaner.

Yep, that's definitely the way to go.

**Pull request merged**